### PR TITLE
Add is_proxied_automattician()

### LINKED ISSUE #314: Update is_automattican requirements
We have a proxy check now. I think we should remove the email verification step and either:

* Add a proxy check to `is_automattican()`; or
* Add an `is_proxied_automattician()` function

I'm leaning toward adding a proxy check directly to `is_automattician()`, but could maybe be convinced otherwise.

### PR DESCRIPTION
**Pull request by joshbetz**
Now that we can make a proxy check, we should add a function which
checks if the current user is proxied. Instead of adding this directly
to is_automattician() like I originally thought, we're introducing a new
function because is_automattician() can also check other users.

Fixes #314

### CODE REVIEW

File: 000-vip-init.php [hunk 1]
```diff
@@ -4,6 +4,10 @@
 	require __DIR__ . '/.secrets/vip-secrets.php';
 }
 
+if ( ! defined( 'A8C_PROXIED_REQUEST' ) ) {
+	define( 'A8C_PROXIED_REQUEST', false );
```
Comment (nickdaugherty) [inline 000-vip-init.php]: I'm worried that it's technically possible to define this constant in the user space.

What prevents a user from adding `define( 'A8C_PROXIED_REQUEST', true )` in their `vip-config.php`?

File: vip-helpers/vip-utils.php [hunk 2]
```diff
@@ -1098,15 +1098,22 @@ function is_automattician( $user_id = false ) {
 		return false;
 	}
 
-	// $vip_support = WPCOM_VIP_Support_User::init();
-
-	if ( WPCOM_VIP_Support_User::is_verified_automattician( $user->ID ) ) {
-		return true;
+	if ( A8C_PROXIED_REQUEST && WPCOM_VIP_Support_User::is_a8c_email( $user->user_email ) ||
```
Comment (nickdaugherty) [inline vip-helpers/vip-utils.php]: Mixing an 'automattician' check and a proxied 'bypass' here seems a little messy. What's the benefit to `is_automattician()` returning true if you're not 'verified', but are proxied?

I think if we want to take the proxy status into account, we should be using `is_proxied_automattician()`, and `is_automattician()` strictly checks if this user is really an Automattician. As is, this introduces some variability into what actually constitutes an Automattician to the system, which is harder to grok. IMO they should be separate concerns unless we require you to be proxied for `is_automattician()` to return `true` in all cases, which I don't think we want.

File: 000-vip-init.php (hunk 1 above)
Comment (joshbetz) [inline 000-vip-init.php]: @nickdaugherty Gooood question. In production, it is always defined in `vip-secrets.php`, so I think you'd generate a fatal if you try to set it in a client repo.

**Comment by joshbetz** [comment]
@nickdaugherty fixed

**Pull request merged**